### PR TITLE
Add/update testing architecture (TestPlay, TestRole, etc.)

### PR DESCRIPTION
**Pull request by alexander-de-ranitz**
Updated outdated elements of existing test files and added TestSkill.

TestPlay, TestRole, etc. are useful for testing purposes (duh), but they were outdated and not in line with the current STP design. Also, there was no TestSkill, so that was added as well in case it is needed during testing. 

###### Code Quality
- [ ] Is the code is understandable and easy to read
- [ ] Changes to the code comply with set clang-format rules
- [ ] No use of manual memory control (e.g new/malloc/colloc etc)
- [ ] Are (only) smart pointers used?

###### Testing
- [ ] All tests are passing.
- [ ] I _added new / changed existing_ tests to reflect code changes (state why not otherwise!)
- [ ] I tested my changes manually (Describe how, to what extent etc.)

###### Commit Messages
- [ ] Commit message is saying what has been changed, **why** it was changed? Remember other developers might not know
  what the problem you are fixing was. Note also negative _decision_ (e.g., why did you not do particular thing)
  **TLDR: Commit message are comprehensive**
- [ ] Commit messages follows the rules of https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -237,7 +238,7 @@ set(INTERFACE_HEADERS
         ${PROJECT_SOURCE_DIR}/include/roboteam_ai/interface/widgets/ManualControlWidget.h
         ${PROJECT_SOURCE_DIR}/include/roboteam_ai/interface/widgets/PlaysWidget.hpp
         ${PROJECT_SOURCE_DIR}/include/roboteam_ai/interface/widgets/GraphWidget.h
-        )
+        include/roboteam_ai/stp/skills/TestSkill.h)
```
Comment (Florisheinen1) [inline CMakeLists.txt]: Please try to use absolute paths with the help of variables in CMake, so if a folder structure ever changes, the CMakeLists files can be easily updated
```suggestion
        ${PROJECT_SOURCE_DIR}/include/roboteam_ai/stp/skills/TestSkill.h)
```

**Review by Florisheinen1 — APPROVED** [review]
Looks fine, compiles. I assume this does not influence the rest of the code, as this just adds mainly example code.

File: CMakeLists.txt (hunk 1 above)
Comment (alexander-de-ranitz) [reply CMakeLists.txt]: Good catch, that include did not belong there, not sure how it got there but it's gone now.

**Review by itsasimpleplan — APPROVED** [review]
Looks good and consistent

**Pull request merged**